tests/Testcontainers.Platform.Linux.Tests/DependsOnTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace Testcontainers.Tests;
+
+public sealed class DependsOnTests : IAsyncLifetime
+{
+    private const string DependsOnKey = "org.testcontainers.depends-on";
+
+    private const string DependsOnValue = "true";
+
+    private static readonly IDictionary<string, bool> LabelFilter = new Dictionary<string, bool> { { string.Join("=", DependsOnKey, DependsOnValue), true } };
+
+    private readonly IContainer _container = new ContainerBuilder()
+        .DependsOn(new ContainerBuilder()
+            .WithImage(CommonImages.Alpine)
+            .WithLabel(DependsOnKey, DependsOnValue)
+            .Build())
+        .DependsOn(new ContainerBuilder()
+            .WithImage(CommonImages.Alpine)
+            .WithLabel(DependsOnKey, DependsOnValue)
+            .Build())
+        .WithNetwork(new NetworkBuilder()
+            .WithLabel(DependsOnKey, DependsOnValue)
+            .Build())
+        .WithVolumeMount(new VolumeBuilder()
+            .WithLabel(DependsOnKey, DependsOnValue)
+            .Build(), "/workdir")
```

Comment:
@cristianrgreco @eddumelendez @mdelapenya @kiview I am interested in what do you think about the following API. It resolves dependent resources internally and creates or starts them before starting the actual container. I am unsure whether we should also replace `WithNetwork(INetwork)` and `WithVolumeMount(IVolume, string)` with `DependsOn(INetwork)`, `DependsOn(IVolume, string)` to achieve a more consistent API. Developers no longer need to create resources upfront.